src/error/error.rs

```diff
@@ -51,83 +88,153 @@ impl error::Error for Error {
     }
 }
 
-impl From<ErrorKind> for Error {
-    fn from(kind: ErrorKind) -> Error {
-        Error {
-            kind,
-        }
-    }
-}
-
 impl fmt::Debug for Error {
     fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
         fmt.debug_struct("Error")
             .field("kind", &self.kind)
+            .field("title", &self.title)
+            .field("detail", &self.detail)
+            .field("status", &self.status)
             .finish()
     }
 }
 
 impl fmt::Display for Error {
     #[allow(deprecated)] // .cause() is deprecated on nightly
     fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
-        use std::error::Error;
+        write!(
+            fmt,
+            "[{}] {}",
+            self.kind, self.title,
+        )?;
+
+        if let Some(ref detail) = self.detail {
+            write!(fmt, ": {}", detail)?;
+        }
 
-        if let Some(cause) = self.cause() {
-            write!(fmt, "{}: {}", self.description(), cause)
-        } else {
-            fmt.write_str(self.description())
+        use std::error::Error;
+        if let Some(ref cause) = self.cause() {
+            write!(fmt, ": {}", cause)?;
         }
+
+        Ok(())
+    }
+}
+
+// Obsolete
+
+/// A list specifying the general categories of Tower Web errors.
+pub struct ErrorKind {
+    kind: KindPriv,
+}
+
+#[derive(Debug, Eq, PartialEq, Clone, Copy)]
+enum KindPriv {
+    BadRequest,
+    Unauthorized,
+    Fordidden,
+    NotFound,
+    UnprocessableEntity,
+    Internal,
+}
+
+impl From<ErrorKind> for Error {
+    fn from(error_kind: ErrorKind) -> Error {
+        Error::blank(error_kind.status_code())
     }
 }
 
 // ===== impl ErrorKind =====
 
 impl ErrorKind {
     /// Returns a new `ErrorKind` value representing a 400 -- bad request error.
+    #[deprecated(note="please use 'Error::blank(http::status::StatusCode::BAD_REQUEST)' instead")]
     pub fn bad_request() -> ErrorKind {
         ErrorKind { kind: BadRequest }
     }
 
     /// Returns `true` if `self` represents a 400 -- bad request error
+    #[deprecated(note="please use 'kind() == http::status::StatusCode::BAD_REQUEST' instead")]
```

Comment:
Do we want to deprecate the `is_*` functions? They still seem useful compared to the alternative.

Reply:
They aren't used in the project at the moment. I've deprecated them in order to eliminate `ErrorKind` itself. We can kill them now and think if we need something similar for `Error` when there is will be a use case :-)

Reply:
Ah, I see it is on `ErrorKind`.